packages/sitecore-jss/src/personalize/utils.test.ts

```diff
@@ -234,42 +243,83 @@ describe('utils', () => {
       });
       it('should use lowercase', () => {
         const pageId = '3E0A2F20B3255E57881FFF6648D08575';
         const language = 'EN';
-        const result = CdpHelper.getContentId(pageId, language);
+        const result = CdpHelper.getPageFriendlyId(pageId, language);
         expect(result).to.equal(`embedded_${pageId}_${language}`.toLowerCase());
       });
       it('should convert language dashes to underscores', () => {
         const pageId = '3E0A2F20B3255E57881FFF6648D08575';
         const language = 'da-DK';
-        const result = CdpHelper.getContentId(pageId, language);
+        const result = CdpHelper.getPageFriendlyId(pageId, language);
         expect(result).to.equal(`embedded_${pageId}_da_DK`.toLowerCase());
       });
       it('should ensure GUID format N for pageId', () => {
         const pageId = '{FFCD3AC4-38E3-5286-A0B9-5F7113D5E74A}';
         const language = 'en';
-        const result = CdpHelper.getContentId(pageId, language);
+        const result = CdpHelper.getPageFriendlyId(pageId, language);
         expect(result).to.equal(
           `embedded_FFCD3AC438E35286A0B95F7113D5E74A_${language}`.toLowerCase()
         );
       });
     });
 
+    describe('getComponentFriendlyId', () => {
+      it('should format', () => {
+        const pageId = '110d559fdea542ea9c1c8a5df7e70ef9';
+        const componentId = '733995bb878042daa6d6c2c5f3f70707';
+        const language = 'en';
+        const result = CdpHelper.getComponentFriendlyId(pageId, componentId, language);
+        expect(result).to.equal(`component_${pageId}_${componentId}_${language}*`);
+      });
+      it('should format with scope', () => {
+        const pageId = '110d559fdea542ea9c1c8a5df7e70ef9';
+        const componentId = '733995bb878042daa6d6c2c5f3f70707';
+        const language = 'en';
+        const scope = 'myscope1';
+        const result = CdpHelper.getComponentFriendlyId(pageId, componentId, language, scope);
+        expect(result).to.equal(`component_${scope}_${pageId}_${componentId}_${language}*`);
+      });
+      it('should use lowercase', () => {
+        const pageId = '3E0A2F20B3255E57881FFF6648D08575';
+        const componentId = '733995BB878042DAA6D6C2C5F3F70707';
+        const language = 'EN';
+        const result = CdpHelper.getComponentFriendlyId(pageId, componentId, language);
+        expect(result).to.equal(`component_${pageId}_${componentId}_${language}*`.toLowerCase());
+      });
+      it('should convert language dashes to underscores', () => {
+        const pageId = '3E0A2F20B3255E57881FFF6648D08575';
+        const componentId = '733995BB878042DAA6D6C2C5F3F70707';
+        const language = 'da-DK';
+        const result = CdpHelper.getComponentFriendlyId(pageId, componentId, language);
+        expect(result).to.equal(`component_${pageId}_${componentId}_da_DK*`.toLowerCase());
+      });
+      it('should ensure GUID format N', () => {
```

Comment:
typo here?

Reply:
Looks like a typo, but intentional. "N" is a GUID format (albeit probably more of a MSFT/.NET thing). https://stackoverflow.com/a/7775466